sdk/core/azure-core/tests/perf_tests/upload_binary.py

```diff
@@ -0,0 +1,94 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+from time import time
+from wsgiref.handlers import format_date_time
+from devtools_testutils.perfstress_tests import RandomStream, AsyncRandomStream
+
+from azure.core.rest import HttpRequest
+from azure.core.exceptions import (
+    HttpResponseError,
+    map_error,
+)
+from azure.core.pipeline.transport import (
+    RequestsTransport,
+    AsyncioRequestsTransport,
+)
+from ._test_base import _BlobTest
+
+import logging
+import sys
+handler = logging.StreamHandler(stream=sys.stdout)
+logger = logging.getLogger('azure')
+logger.setLevel(logging.DEBUG)
+logger.addHandler(handler)
+
+class UploadBinaryDataTest(_BlobTest):
+    def __init__(self, arguments):
+        super().__init__(arguments)
+        blob_name = "uploadtest"
+        self.blob_endpoint = f"{self.account_endpoint}{self.container_name}/{blob_name}"
+        self.upload_stream = RandomStream(self.args.size)
+        self.upload_stream_async = AsyncRandomStream(self.args.size)
+        # TODO: investigate size-1 issue for requests transports - https://github.com/Azure/azure-sdk-for-python/issues/33997
+        if self.sync_transport == RequestsTransport:
+            self.sync_size = self.args.size - 1
+        else:
+            self.sync_size = self.args.size
+        if self.async_transport == AsyncioRequestsTransport:
+            self.async_size = self.args.size - 1
+        else:
+            self.async_size = self.args.size
+
+    # TODO: If seeing "the request verb is invalid" error frequently: https://github.com/Azure/azure-sdk-for-python/issues/33999
```

Comment:
It's possible one or more of these errors are because we are sharing the stream (self.upload_stream) which may have an impact when these tests are run in parallel and the stream is being accessed from multiple threads. Why that would present as an invalid verb error I have no idea.... but it may still cause an issue :D

Reply:
will try specifying parallel=1 and see if that helps anything...

Reply:
resolved here: https://github.com/Azure/azure-sdk-for-python/issues/33999